File: python/fasta/fasta.py
import random

def generate_dna_sequence(length):
    """Generate a random DNA sequence of given length."""
    return ''.join(random.choice('ACGT') for _ in range(length))

def write_fasta(sequences, filename):
    """Write sequences to a FASTA file."""
    with open(filename, 'w') as file:
        for i, seq in enumerate(sequences, start=1):
            file.write(f">seq{i}\n{seq}\n")

def main():
    num_sequences = 5  # Number of sequences to generate
    sequence_length = 50  # Length of each sequence
    fasta_filename = "random_dna_sequences.fasta"
    
    sequences = [generate_dna_sequence(sequence_length) for _ in range(num_sequences)]
    write_fasta(sequences, fasta_filename)
    print(f"Generated {num_sequences} random DNA sequences of length {sequence_length} and saved to {fasta_filename}")

if __name__ == "__main__":
    main()

File: c/fasta/fasta.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define SEQUENCE_LENGTH 50
#define NUM_SEQUENCES 5

/* Function to generate a single random DNA base */
char generate_random_base() {
    char bases[] = {'A', 'C', 'G', 'T'};
    return bases[rand() % 4]; // Randomly choose a base
}

/* Function to generate a random DNA sequence */
void generate_dna_sequence(char *sequence, int length) {
    for (int i = 0; i < length; i++) {
        sequence[i] = generate_random_base();
    }
    sequence[length] = '\0'; // Null-terminate the string
}

/* Function to write sequences to a FASTA file */
void write_fasta(char sequences[NUM_SEQUENCES][SEQUENCE_LENGTH + 1], const char *filename) {
    FILE *file = fopen(filename, "w");
    if (file == NULL) {
        fprintf(stderr, "Error opening file %s for writing\n", filename);
        exit(1);
    }

    for (int i = 0; i < NUM_SEQUENCES; i++) {
        fprintf(file, ">seq%d\n", i + 1);
        fprintf(file, "%s\n", sequences[i]);
    }

    fclose(file);
}

int main() {
    char sequences[NUM_SEQUENCES][SEQUENCE_LENGTH + 1]; // +1 for the null terminator
    srand(time(NULL)); // Seed the random number generator

    // Generate random DNA sequences
    for (int i = 0; i < NUM_SEQUENCES; i++) {
        generate_dna_sequence(sequences[i], SEQUENCE_LENGTH);
    }

    // Write the sequences to a FASTA file
    write_fasta(sequences, "random_dna_sequences.fasta");

    printf("Generated %d random DNA sequences of length %d and saved to random_dna_sequences.fasta\n", NUM_SEQUENCES, SEQUENCE_LENGTH);
    return 0;
}
